compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -195,8 +290,35 @@ struct GPUSetEncodingOpLoweringConversion
     //    LHS: 64 -> 16x4 (innerTiles[0]xinnerTiles[1])
     //    ACC: 256 -> 16x16 (innerTiles[0]xinnerTiles[1])
 
-    // TODO(lialan): Replace the op with the tensor.expand_shape op.
-    rewriter.replaceOp(encodingOp, packOp->getResult());
+    // collapse tiled dimensions into one dim
+    SmallVector<ReassociationIndices> collapseReassoc = {
+        {0}, {1}, {2, 3, 4, 5}};
+    SmallVector<int64_t> collapsedShape = {sourceShape[0], sourceShape[1],
+                                           sourceShape[2] * sourceShape[3]};
+    RankedTensorType revertShapeType = RankedTensorType::get(
+        collapsedShape, encodingOp.getSourceType().getElementType());
+
+    auto collapseShapeOp = rewriter.create<tensor::CollapseShapeOp>(
+        loc, revertShapeType, transposeOp->getResult(0),
+        ArrayRef<ReassociationIndices>(collapseReassoc));
+
+    // expand the collapsed shape to the shape intended by the encoding
+    // Create expand_shape, to tile the inner most two dimensions.
+    assert(innerTiles.size() == 2); // TODO: relax this
+    assert(innerTiles[0] * innerTiles[1] == sourceShape[2] * sourceShape[3]);
+    RankedTensorType expandTileShapeType = RankedTensorType::get(
+        {sourceShape[0], sourceShape[1], innerTiles[0], innerTiles[1]},
+        encodingOp.getSourceType().getElementType());
+    SmallVector<ReassociationExprs> tileAssoc(3);
+    tileAssoc[0].push_back(rewriter.getAffineDimExpr(0));
+    tileAssoc[1].push_back(rewriter.getAffineDimExpr(1));
+    tileAssoc[2].push_back(rewriter.getAffineDimExpr(2));
+    tileAssoc[2].push_back(rewriter.getAffineDimExpr(3));
```

Comment:
nit: we can simplify the code using `SmallVector<ReassociationIndices>`, like what you've done in l.294.

Reply:
True, made life much simpler